scripts/ci-entrypoint.sh

```diff
@@ -93,6 +97,19 @@ create_cluster() {
     ${REPO_ROOT}/hack/create-dev-cluster.sh
 }
 
+wait_for_nodes() {
+    echo "Waiting for ${CONTROL_PLANE_MACHINE_COUNT} control plane machine and ${WORKER_MACHINE_COUNT} worker machine to become Ready"
+
+    # Ensure that all nodes are registered with the API server before checking for readiness
+    local total_nodes="$((${CONTROL_PLANE_MACHINE_COUNT} + ${WORKER_MACHINE_COUNT}))"
+    while [[ $(kubectl get nodes -ojson | jq '.items | length') -ne "${total_nodes}" ]]; do
+        sleep 10
+    done
+
+    kubectl wait --for=condition=Ready node --all --timeout=5m
```

Comment:
how does this pass given #656 ?

Reply:
It took 15 minutes for the machinepool to register with the control plane. That's why I have a 30 minutes timeout in https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/659/files#diff-4bc72f6db52f65045884231f54b3e802R105-R107

Reply:
does that mean the e2e tests will take that much longer?

Reply:
It should take the same amount of time. This is added so that E2E tests don't have to implement scripts on their side to wait for worker machines to become Ready before starting.